packages/mdc-feature-targeting/_functions.scss

```diff
@@ -0,0 +1,170 @@
+@import "./variables";
+
+// ==Terminology==
+// Feature:
+//   A simple string (e.g. `color`) representing a cross-cutting feature in
+//   Material.
+// Feature query:
+//   A structure that represents a query for a feature or combination of features. This may be
+//   either a feature or a map containing `op` and `queries` fields. A single feature represents a
+//   simple query for just that feature. A map represents a complex query made up of an operator,
+//   `op`, applied to a list of sub-queries, `queries`.
+//   (e.g. `color`, `(op: any, queries: (color, typography))`).
+// Feature target:
+//   A map that contains the feature being targeted as well as the current feature query. This is
+//   the structure that is intended to be passed to the `@mdc-feature-targets` mixin.
+//   (e.g. `(target: color, query: (op: any, queries: (color, typography))`).
+
+//
+// Public
+//
+
+// Creates a feature target from the given feature query and targeted feature.
+@function mdc-feature-create-target($feature-query, $targeted-feature) {
+  $feature-target: (target: $targeted-feature, query: $feature-query);
+  $valid: mdc-feature-verify-target_($feature-target);
+
+  @return $feature-target;
+}
+
+// Parses a list of feature targets to produce a map containing the feature query and list of
+// available features.
+@function mdc-feature-parse-targets($feature-targets) {
+  $valid: mdc-feature-verify-target_($feature-targets...);
+  $available-features: ();
+
+  @each $target in $feature-targets {
+    $available-features: append($available-features, map-get($target, target));
+  }
+
+  @return (
+    available: $available-features,
+    query: map-get(nth($feature-targets, 1), query)
+  );
+}
+
+// Creates a feature query that is satisfied iff all of its sub-queries are satisfied.
+@function mdc-feature-all($feature-queries...) {
+  $valid: mdc-feature-verify-query_($feature-queries...);
```

Comment:
I think I wrote this originally. I remember it either stripping it or failing some kind of lint check because nothing was being done with the result. Can't remember which...

Reply:
I gave these the hairy eyeball at first as well. It's presumably related to Sass functions being required to return values, but these function calls are used purely in an assertion sense - they throw errors if they fail.

Reply:
gotcha...its just some sass issue